### PR TITLE
[Docs] Add contribution guides

### PR DESCRIPTION
**Pull request by gaotongxiao**

### CODE REVIEW

File: docs/en/notes/contribution_guide.md [hunk 1]
```diff
@@ -1 +1,136 @@
...
+  Checkout the latest branch of the local repository and pull the latest branch of the source repository. Here we assume that you are working on the `dev-1.x` branch.
+
+  ```bash
+  git checkout dev-1.x
+  git pull upstream dev-1.x
+  ```
+
+### 2. Checkout a new branch from the `main` branch or `dev-1.x` branch
+
+````bash
```
Comment (xinke-wang) [inline docs/en/notes/contribution_guide.md:54]:
```suggestion

```

File: docs/en/notes/contribution_guide.md [hunk 2]
```diff
@@ -1 +1,136 @@
...
+
+```bash
+git checkout -b branchname
+````
+
+```{tip}
+To make commit history clear, we strongly recommend you checkout the `main`/`dev-1.x` branch before creating a new branch.
```
Comment (xinke-wang) [inline docs/en/notes/contribution_guide.md:61]:
```suggestion
To make commit history clear, we strongly recommend you checkout the `main` or `dev-1.x` branch before creating a new branch.
```

image not clear

File: docs/zh_cn/notes/contribution_guide.md [hunk 3]
```diff
@@ -1 +1,134 @@
...
+  ```
+
+### 2. 从 `main` 或 `dev-1.x` 分支创建一个新的开发分支
+
+```bash
+git checkout -b branchname
+```
+
+```{tip}
+为了保证提交历史清晰可读，我们强烈推荐您先检出 `main`/`dev` 分支，再创建新的分支。
```
Comment (xinke-wang) [inline docs/zh_cn/notes/contribution_guide.md:59]:
```suggestion
为了保证提交历史清晰可读，我们强烈推荐您先切换到 `main` 或 `dev` 分支，再创建新的分支。
```

File: docs/zh_cn/notes/contribution_guide.md [hunk 4]
```diff
@@ -1 +1,134 @@
...
+- 提交修改。在每次提交前，pre-commit hooks 都会被触发并规范化你的代码格式。
+
+  ```bash
+  # coding
+  git add [files]
+  git commit -m 'messages'
+  ```
+
+  ```{note}
+  有时你的文件可能会在提交时 pre-commit hooks 自动修改。这时请重新添加并提交修改后的文件。
```
Comment (xinke-wang) [inline docs/zh_cn/notes/contribution_guide.md:80]:
```suggestion
  有时你的文件可能会在提交时被 pre-commit hooks 自动修改。这时请重新添加并提交修改后的文件。
```

**Comment by xinke-wang** [comment]
And don't forget to revise the link in README.md

image

**Comment by gaotongxiao** [comment]
@xinke-wang The link in readme still works. `.github/CONTRIBUTING.md` has been revised accordingly and can point users to the contribution guide.

**Review by xinke-wang — APPROVED** [review]

**Pull request merged**